GPflow/param.py

```diff
@@ -284,6 +284,30 @@ def set_state(self, x):
         self._array[...] = new_array
         return free_size
 
+    def randomize(self, distributions={}, skipfixed=True):
+        """
+        Randomly assign the parameter a new value by sampling either from a
+        provided distribution from GPflow.priors, the parameter's prior, or
+        by using a default scheme where a standard normal variable is
+        propagated through the parameters transform.
+        Will not change fixed parameters unless skipfixed flag is set to False.
+
+        Optional Input:
+            distributions (dictionary) - a list of priors indexed by parameters.
+                Defaults to an empty dictionary.
+            skipfixed (boolean) - if True, parameter cannot be randomized.
+                Defaults to True.
+        """
+        if not (skipfixed and self.fixed):
+            if self in distributions.keys():
+                self._array = distributions[self].sample(self.shape)
+            else:
+                try:
+                    self._array = self.prior.sample(self.shape)
+                except AttributeError:
+                    randn = np.array(np.random.randn(*self.shape))
```

Comment:
Some transforms alter the size of their argument. I think you need `self.transform.free_state_size(self.shape)` to get the correct number of randns here.